### PR TITLE
unblock lingui and allow for SSR on almost all pages

### PR DESCRIPTION
**Pull request by wraeth-eth**
## What does this PR do and why?

The big kahuna - this breaks lingui out into SSR and will not block for client side compilation/rendering

## Screenshots or screen recordings

_If applicable, provide screenshots or screen recordings to demonstrate the changes._

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] (if relevant) I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] (if relevant) I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

**Comment by wraeth-eth** [comment]
Current dependencies on/for this PR:
* main
  * **PR #3997** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3997" target="_blank">Graphite</a> 
    * **PR #3998** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3998" target="_blank">Graphite</a> 
      * **PR #3999** <a href="https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3999" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3999?utm_source=stack-comment).

File: src/components/ErrorNotificationButtons.tsx [hunk 1]
```diff
@@ -14,20 +12,16 @@ const resetSite = () => {
...
-          </span>
-        </Button>
-      </div>
-    </LanguageProvider>
+    <div className="flex items-center gap-5">
+      <ExternalLink
+        className="text-grey-900 hover:underline dark:text-slate-100"
+        href={helpPagePath('/user/resources/troubleshoot/')}
+      >
+        Troubleshoot
```
[the remark this answers is not in the document]
Comment (wraeth-eth) [reply src/components/ErrorNotificationButtons.tsx]: Fixed it! - no longer a casualty, brought it back!

File: package.json [hunk 2]
```diff
@@ -160,6 +160,8 @@
     "@graphql-codegen/typescript-operations": "^3.0.3",
     "@graphql-codegen/typescript-react-apollo": "^3.3.2",
     "@graphql-codegen/typescript-resolvers": "^3.2.0",
+    "@lingui/loader": "^4.4.0",
+    "@lingui/swc-plugin": "^4.0.4",
```
Comment (tomquirk) [inline package.json:164]: just gotta be realll careful with this version. when i last looked into this, it was a very specific combo of next, lingui and swc_core. https://github.com/lingui/swc-plugin#compatibility

we should prob fix this version, wdyt

File: src/components/Navbar/components/NavLanguageSelector.tsx [hunk 3]
```diff
@@ -33,15 +26,19 @@ export default function NavLanguageSelector({
...
       items={Object.values(SUPPORTED_LANGUAGES).map(lang => ({
         id: lang.code,
         label: lang.long,
-        onClick: () => {
-          setLanguage(lang.code)
+        // TODO: We want to use the bottom but due to a bug in t macros we cant
+        // locale: lang.code,
+        // href: pathname,
+        onClick: async () => {
+          await router.push(router.asPath, router.asPath, { locale: lang.code })
```
Comment (tomquirk) [inline src/components/Navbar/components/NavLanguageSelector.tsx:40]: cleannn

File: src/pages/about/index.tsx [hunk 4]
```diff
@@ -17,3 +18,5 @@ export default function AboutPage() {
     </>
   )
 }
+
+export const getServerSideProps = globalGetServerSideProps
```
[the remark this answers is not in the document]
Comment (tomquirk) [reply src/pages/about/index.tsx:22]: ah yep, lol. wen approuter

**Review by tomquirk — APPROVED** [review]
@wraeth-eth MEGA! thanks for taking this initiative, will be so great to have this nailed for other JB apps too.

Code LGTM and makes sense. Warning: I have not tested this in prod. Happy to pre approve and test tomorrow but if you're confident, def send it 🚀  (just one comment about the package versioning)

File: package.json (hunk 2 above)
Comment (wraeth-eth) [reply package.json:164]: happy to fix

**Comment by wraeth-eth** [comment]
> @wraeth-eth MEGA! thanks for taking this initiative, will be so great to have this nailed for other JB apps too.
> 
> Code LGTM and makes sense. Warning: I have not tested this in prod. Happy to pre approve and test tomorrow but if you're confident, def send it 🚀 (just one comment about the package versioning)

I think zhape actually ran through and tested this on a preview for me

**Comment by wraeth-eth** [comment]
### Merge Activity

* **Aug 31, 7:04 PM**: @wraeth-eth started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/jbx-protocol/juice-interface/3999).

**Pull request merged**